tests/rpc/rpc_01-system_test.go

```diff
@@ -0,0 +1,181 @@
+
+package rpc
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+	"reflect"
+	"testing"
+	"time"
+
+	"github.com/ChainSafe/gossamer/dot/rpc/modules"
+	"github.com/ChainSafe/gossamer/lib/common"
+	"github.com/stretchr/testify/require"
+)
+
+func TestSystemRPC(t *testing.T) {
+	if GOSSAMER_INTEGRATION_TEST_MODE != rpcSpec {
+		_, _ = fmt.Fprintln(os.Stdout, "Going to skip RPC spec tests")
+		return
+	}
+
+	testsCases := []struct {
+		description string
+		method      string
+		expected    interface{}
+		skip        bool
+	}{
+		{ //TODO
+			description: "test system_name",
+			method:      "system_name",
+			skip:        true,
+		},
+		{ //TODO
+			description: "test system_version",
+			method:      "system_version",
+			skip:        true,
+		},
+		{ //TODO
+			description: "test system_chain",
+			method:      "system_chain",
+			skip:        true,
+		},
+		{ //TODO
+			description: "test system_properties",
+			method:      "system_properties",
+			skip:        true,
+		},
+		{
+			description: "test system_health",
+			method:      "system_health",
+			expected: modules.SystemHealthResponse{
+				Health: common.Health{
+					Peers:           2,
+					IsSyncing:       false,
+					ShouldHavePeers: true,
+				},
+			},
+			skip: false,
+		},
+		{
+			description: "test system_peers",
+			method:      "system_peers",
+			expected: modules.SystemPeersResponse{
+				Peers: []common.PeerInfo{},
+			},
+			skip: false,
+		},
+		{
+			description: "test system_network_state",
+			method:      "system_networkState",
+			expected: modules.SystemNetworkStateResponse{
+				NetworkState: common.NetworkState{
+					PeerID: "",
+				},
+			},
+			skip: false,
+		},
+		{ //TODO
+			description: "test system_addReservedPeer",
+			method:      "system_addReservedPeer",
+			skip:        true,
+		},
+		{ //TODO
+			description: "test system_removeReservedPeer",
+			method:      "system_removeReservedPeer",
+			skip:        true,
+		},
+		{ //TODO
+			description: "test system_nodeRoles",
+			method:      "system_nodeRoles",
+			skip:        true,
+		},
+		{ //TODO
+			description: "test system_accountNextIndex",
+			method:      "system_accountNextIndex",
+			skip:        true,
+		},
+	}
+
+	t.Log("going to start gossamer")
+
+	localPidList, err := StartNodes(t, make([]*exec.Cmd, 1))
+
+	//use only first server for tests
+	require.Nil(t, err)
+
+	time.Sleep(time.Second) // give server a second to start
+
+	for _, test := range testsCases {
+		t.Run(test.description, func(t *testing.T) {
+			if test.skip {
+				t.Skip("RPC endpoint not yet implemented")
+				return
+			}
+
+			respBody, err := PostRPC(t, test.method, "http://"+GOSSAMER_NODE_HOST+":"+currentPort, "{}")
+			require.Nil(t, err)
+
+			target := DecodeRPC(t, respBody, test.method)
+
+			require.NotNil(t, target)
+
+			t.Log("Will start assertion for ", "target", target, "type", reflect.TypeOf(target))
+
+			switch v := target.(type) {
```

Comment:
shouldn't we switch on the test description? what if an rpc method returns the wrong type

Reply:
The RPC method should never return the wrong type, if that is happening we are in deep trouble.

Reply:
okay well this will panic if it returns the wrong type, is that the desired behaviour?

Reply:
Why would it panic ? it just not going to match any type. If something else is going on with the endpoint it will be either captured at unit tests or on DecodeRPC method

Reply:
There is also the require nil err just before the switch...if you happen to be able to prove the edge case you are mentioning I will be more than happy to enforce more rules. Right now it seems like a overkill...

Reply:
looking at the DecodeRPC method, the switch statement seems very brittle and will need to be updated in case the RPC api changes. it would be better to pass in a `target interface{}` directly instead of `targetType string` and switching on it. maybe we could open an issue for that?

Reply:
Well, that one should be always updated. That is why this is going to be part of the CI. 
The trick is that the `DisallowUnknownFields` will make it fail nicely if the endpoint return changes, so we will see the test failures beforehand.